README.md

```diff
@@ -425,6 +432,84 @@ div(
 // </div>
 ```
 
+####  Handler
+Because of Outwatch is following a reactive paradigm, based at this moment in monix.
+
+We have `Observables` and `Observers` from monix.
+
+and `Handler` from Outwatch:
+
+Handler is just an alias for `Observer with Observable` and is referential transparent API. (before was `Sink with Observer` where `Sink` was just a wrapper around Observer). 
+
+So `Handler` creates an `Observable` that is also a `Observer`.
+
+Older version  patterns changes from.
+
+```scala
+val  textValues = createStringHandler()
+```
+
+to 
+
+```scala
+val textValues = Handler.create[String].unsafeRunSync()
+```
+
+and this
+
+```scala
+val additions = createHandler[Int]()
+```
+
+to 
+
+```scala
+val additions = Handler.create[Int].unsafeRunSync()
+```
+
+When is the right time to put the .unsafeRunSync on there?
```

Comment:
It should be noted here, that the whole idea of the handler wrapped in IO is **referential transparency**. If the component is in a `def`, it is already referential transparent, because for each invocation a new `Handler` is created, so the usage `unsafeRunSync()` is fine in this case (@cornerman correct me if I'm wrong). If the same code is stored in a `val` and the `val` is used several times, the created `Handler` is hared between the usages and is therefore not referential transparent.

Reply:
The whole idea of handler being wrapped in an effect type is indeed referential transparency. Though, it does not matter whether the component is in a `def` or a `val`, you should not run `unsafeRunSync` within your app, but in the best case only once in your main, when running `OutWatch.render*`. You can map over `IO[Handler]` if you want to share some handler state, but it will be explicit.

Take for example:

```scala
def counter = Handler.create[Int].map(handler => div(handler, onClick(handler.map(_ + 1)) --> handler)
val counterResult = counter

// the following two line are equivalent:
div(counter, counter)
div(counterResult, counterResult)
```

Now, if you run unsafeRunSync in your counter component, your method will not be referentially transparent:

```scala
def counter = {
  val handler = Handler.unsafe[Int]
  div(handler, onClick(handler.map(_ + 1)) --> handler)
}
val counterResult = counter

// the following two line are NOT equivalent:
div(counter, counter)
div(counterResult, counterResult)
```